src/main/java/com/iota/iri/cache/Cache.java

```diff
@@ -0,0 +1,113 @@
+package com.iota.iri.cache;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Cache operations
+ * 
+ * @param <K> Key type
+ * @param <V> Value type
+ */
+public interface Cache<K, V> {
+
+    /**
+     * Get the value mapped to the specified key.
+     * If it's not found, it reads from DB and updates cache.
+     * 
+     * @param key The keys whose value is to be returned
+     * @return The mapped value of the specified key
+     */
+    V get(K key);
+
+    /**
+     * Gets the value if it is in cache. Does not read from DB if not found.
+     * @param key The key
+     * @return The cached value.
+     */
+    V lookup(K key);
+
+    /**
+     * Get all elements from the cache for the keys specified.
+     * 
+     * @param keys The keys to get values of
+     * @return A map of key and value associations.
+     */
+    Map<K, V> getAll(Collection<K> keys);
+
+    /**
+     * The size of the cache
+     * 
+     * @return The size of the cache
+     */
+    int getSize();
+
+    /**
+     * Associates the specified value with the specified key in the cache. If the cache already contains a value of this
+     * key, the old value will be replaced with this value.
+     * 
+     * @param key   They key to associate the value with
+     * @param value The value to associate the key with
+     */
+    void put(K key, V value);
+
+    /**
+     * Associates the specified value with the specified key if the key does not already exist in the cache If the key
+     * already exists, the existing value is returned.
+     * 
+     * @param key   The key to associate the specified value with
+     * @param value The value to associate the specified key with
+     */
+    void putIfAbsent(K key, V value);
```

Comment:
Does this make anything more efficient?
I suppose you are doing a `get` and then a `put`

Reply:
`putIfAbsent` is only useful if we don't want to update an already existing entry.
I've not used it yet though.

Reply:
I can't imagine a scenario where it will be used